### PR TITLE
PV2-620-datalock-actor-map-earning-event

### PR DESCRIPTION
**Pull request by carlbm**
Datalocks service now returning passthough mapping
Removed readonly list from events (issues with mapping)
Added test project for the datalocks service

### CODE REVIEW

File: src/SFA.DAS.Payments.Datalocks.Messages/DataLockEvent.cs [hunk 1]
```diff
@@ -8,11 +8,11 @@ namespace SFA.DAS.Payments.DataLocks.Messages
 {
     public abstract class DataLockEvent : PaymentsEvent, IContractType1EarningEvent
     {
-        public ReadOnlyCollection<PriceEpisode> PriceEpisodes { get; set; }
+        public List<PriceEpisode> PriceEpisodes { get; set; }
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.Datalocks.Messages/DataLockEvent.cs]: Whiy change? Why to List?

File: src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs [hunk 2]
```diff
@@ -11,14 +12,16 @@ namespace SFA.DAS.Payments.DataLocks.DataLockService
...
-        public DataLockService(ActorService actorService, ActorId actorId): base(actorService, actorId)
+        private readonly IMapper mapper;
+
+        public DataLockService(ActorService actorService, ActorId actorId, IMapper mapper) : base(actorService, actorId)
         {
+            this.mapper = mapper;
         }
 
-        public Task<DataLockEvent> HandlePayment(ApprenticeshipContractType1EarningEvent message, CancellationToken cancellationToken)
+        public async Task<DataLockEvent> HandlePayment(ApprenticeshipContractType1EarningEvent message, CancellationToken cancellationToken)
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: Should this method be called HandleEarning?

Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: Can the return value be a PayableEarningEvent? This result is going to be published by the Proxy service and the receiving service expects a PayableEarningEvent, not a DataLockEvent

File: src/SFA.DAS.Payments.Datalocks.Messages/DataLockEvent.cs (hunk 1 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Datalocks.Messages/DataLockEvent.cs]: Automapper...

File: src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs (hunk 2 above)
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: There is no await in the method. Does this need to be async?

Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: So what happens with a non-payable earning?

Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: It helps with the casting of the generics. May be irrelevant depending on how we deal with DatalockEvent vs PayableEarningsEvent

Comment (davehowlett) [reply src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: good point. At the moment, we just require a PayableEarningEvent, but down the wire the DataLockService will be returning the correct object type that the handler can then just publish (and which will then be routed to the correct place down the line).

Happy to leave it as it is.

Comment (davehowlett) [reply src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs]: interesting - after pulling this branch into PV2-619 the warning about async not being required has disappeared! Happy to leave this

**Comment by carlbm** [comment]
Already part of feature branch

**Pull request closed without merging**